### ISSUE TITLE
Pictures being viewed are automatically being downloaded

### ISSUE THREAD

**Issue by enoch85** [owncloud/android#588]
As it is today, viewing pictures in the Android app also downloads them to the phone memory. It would be much better if the pictures wasn´t saved (would be better to have that as an option in the drop-down menu when long-pressing(?) on the filename) to the internal memory.

I don´t know why they are saved to the internal memory today, but my wild guess is that it has to be saved somewhere in order to be shown in the app. Why not just use a "direct connection" (WebDAV) or store them temporarily while shown, and then deleted when you slide to next picture? This would save both time (don´t have to delete the files that are stored like every week on a phone with lack of memory) and memory of course, as the pictures aren´t saved permanently.

What do you think @davivel ?

**Comment by JasperWeiss** [comment]
I still feel this isn't solved. When using dropbox, onedrive etc. I can browse through files and download them to my phone if I need to have it available offline. Now everytime I view a file it's stored permanently on my device and I have to manually delete it. All cloud storage apps have a separate view and download option. Why is it so hard to make this work properly with owncloud

**Comment by enoch85** [comment]
As I agree with @JasperWeiss I reopen this issue. Still, nothing has happened to solve this issue, or if so, please link the PR in this issue.

**Comment by LukeOwncloud** [comment]
There is something happening. The the beta version provided by Tobias (branch beta). It does not load the whole file but a scaled version.

**Comment by enoch85** [comment]
@LukeOwncloud How do I get the Beta version so that I can try it?

**Comment by JasperWeiss** [comment]
@LukeOwncloud the problem isn't that I have little bandwith or storage space. It's just that the idea of cloud storage is to store things in the cloud. Now, everytime I view a file, it downloads it and opens it. After which I have to remove the local copy. I just want to tap a file, view the image/document/song/video without having it downloaded to my device first (at least not permanently, there's always going to be a need for a cache/temorary file)

**Comment by enoch85** [comment]
Just to take one example.

You could use the Gallery app on the server for viewing pictures, that works exactly as it should, it shows the picture, and you can *choose* to download it. 

But imagine if it had the same behaviour as the android app, "to save bandwidth" or "always make the file available" it would download the picture to the computer every time. Now, wouldn't that be a pretty bad solution?

**Comment by JasperWeiss** [comment]
Exactly. I'd say go ahead and check out Onedrive, dropbox, Google drive. And see how they handle things. 

Side note, I am planning to make my own app that uses the onedrive/google drive/dropbox api to access files so that you can use it the same way the official apps do, except it encrypts it first before sending it. This allows you to use any cloud service provider and have the owncloud like peace of mind that nobody can access your files except you. (Basically boxcryptor but open-source with a intuitive material interface)

**Comment by LukeOwncloud** [comment]
@enoch85 
The beta is here: https://github.com/owncloud/android/tree/beta/apks
Direct link to APK: https://github.com/owncloud/android/raw/beta/apks/latest.apk

@enoch85 @JasperWeiss 
I do not argue with you guys and I would not have the slightest problem with such a setting. But I think it is not necessary...

@tobiasKaminsky 
...because the beta can do exactly what you want, I think. Can you shed some light on how the scaled photos are cached? Cache size? Set cache size? To 0?

@JasperWeiss 
That seems to me a mammoth task. Good luck!

**Comment by enoch85** [comment]
@LukeOwncloud 

>  But I think it is not necessary...

Not for you maybe, and not for me neither really (I have 64 GB phone) but I'm thinking of all the other users with smaller memory. ownCloud is for everybody. Just sayin. :)

**Comment by JasperWeiss** [comment]
@luke @enoch85 again, for me too it's not so much about storage space. It's just rather annoying behavior for a cloud storage app to download every file you open.

**Comment by JasperWeiss** [comment]
Oh and @lukeOwncloud, it shouldn't be too hard since there are plenty of file managers that I can borrow code from and all the connection stuff is handled by the api.
I'm just trying to think of a way to have the encryption key on multple devices without manually transferring it. I guess it should be stored on the cloud service along with the files encrypted with a (strong) password but that wouldn't be ideal.

**Comment by enoch85** [comment]
@LukeOwncloud Wow, the Beta version is really nice. Afaik the file is  not stored locally if you choose "do nothing", and the file is not downloaded when you browse the pictures on server. Is that a correct assumption?

@tobiasKaminsky Will this be in 1.8.1?

**Comment by LukeOwncloud** [comment]
@JasperWeiss 
> It's just rather annoying behavior for a cloud storage app to download every file you open.

The beta does not (at least in the context of photo).

@enoch85 
No, "do nothing" is an action for instant upload (observer the title of the section).

Whole photos a never downloaded unless you explicitly choose from the menu ``download``. As I wrote before:
> [The beta version] does not load the whole file but a scaled version. 

What you want to do is go into the settings and set the cache size to 0MB. Then all scaled images are delete right after they are loaded.

@tobiasKaminsky That is what actually happens (I just checked). However, would it be better to clean up the cache only when changing the folder? Otherwise thumbnails do not make any sense: They would be gone after scrolling down and up again.
